extensions/wikia/Search/classes/Result/ResultHelper.php

```diff
@@ -1,67 +1,90 @@
 		$imageURL = ImagesService::getImageSrc(
 			$result['id'],
 			$commData->getCommunityImageId(),
-			WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_WIDTH,
-			WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_HEIGHT)['src'];
+			$imageSizes['width'],
+			$imageSizes['height']
+		)['src'];
 
 		$thumbTracking = "thumb";
 		//Fallback: if Curated Mainpage is inaccessible, try to use Special:Promote
 		//TODO: Remove after DAT-3642 is done
-		if (empty($imageURL)) {
-			$imageFileName = PromoImage::fromPathname($result['image_s'])->ensureCityIdIsSet($result['id'])->getPathname();
+		if ( empty( $imageURL ) ) {
+			$imageFileName = PromoImage::fromPathname( $result['image_s'] )
+				->ensureCityIdIsSet( $result['id'] )
+				->getPathname();
 			$imageURL = ImagesService::getImageSrcByTitle(
-				(new \CityVisualization)->getTargetWikiId($result['lang_s']),
+				( new \CityVisualization)->getTargetWikiId( $result['lang_s'] ),
 				$imageFileName,
-				WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_WIDTH,
-				WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_HEIGHT);
+				$imageSizes['width'],
+				$imageSizes['height']
+			);
 		}//TODO: end
 
-		if (empty($imageURL)) {
+		if ( empty( $imageURL ) ) {
 			// display placeholder image if no thumbnail
 			$imageURL = \F::app()->wg->ExtensionsPath . '/wikia/Search/images/wiki_image_placeholder.png';
 			$thumbTracking = "no-thumb";
 		}
 
-		$description = $result->limitTextLength($commData->getCommunityDescription(), $descWordLimit);
-		$description = !empty($description)
+		$description = $result->limitTextLength( $commData->getCommunityDescription(), $descWordLimit );
+		$description = !empty( $description )
 			? $description
-			: $result->getText(Utilities::field('description'), $descWordLimit);
+			: $result->getText( Utilities::field( 'description' ), $descWordLimit );
 
 		$service = new MediaWikiService();
+		$wikiaSearchHelper = new \WikiaSearchHelper();
+
+		$globalSearchUrl = '';
+		if ( $query ) {
+			$lang = $wikiaSearchHelper->getLangForSearchResults();
+			$centralUrl = $wikiaSearchHelper->getCentralUrlFromGlobalTitle( $lang );
+			$globalSearchUrl = $wikiaSearchHelper->getGlobalSearchUrl( $centralUrl ) . '?search='. $query;
+		}
 
 		return [
-			'isOnWikiMatch' => isset($result['onWikiMatch']) && $result['onWikiMatch'],
+			'isOnWikiMatch' => isset( $result['onWikiMatch'] ) && $result['onWikiMatch'],
 			'imageURL' => $imageURL,
 			'description' => $description,
-			'pagesMsg' => $service->shortnumForMsg($result['articles_i'] ?: 0, 'wikiasearch2-pages'),
-			'imgMsg' => $service->shortnumForMsg($result['images_i'] ?: 0, 'wikiasearch2-images'),
-			'videoMsg' => $service->shortnumForMsg($result['videos_i'] ?: 0, 'wikiasearch2-videos'),
-			'title' => ($sn = $result->getText('sitename_txt')) ? $sn : $result->getText('headline_txt'),
-			'url' => $result->getText('url'),
+			'pages' => [
+				'count' => $service->shortNumForMsg( $result['articles_i'] ?: 0 ),
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "count" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)